whisper/src/message.rs

```diff
@@ -44,68 +71,69 @@ pub fn work_factor_proved(size: u64, ttl: u64, hash: H256) -> f64 {
 	2.0_f64.powi(leading_zeros as i32) / spacetime
 }
 
-/// A topic of a message.
+/// A topic of a message. The topic is an abridged version of the first four bytes of the original topic's hash.
 #[derive(Debug, Default, Clone, Copy, Hash, PartialEq, Eq, PartialOrd, Ord)]
-pub struct Topic(pub [u8; 4]);
+pub struct EnvelopeTopic(pub EnvelopeTopicsData);
 
-impl From<[u8; 4]> for Topic {
-	fn from(x: [u8; 4]) -> Self {
-		Topic(x)
+impl From<EnvelopeTopicsData> for EnvelopeTopic {
+	fn from(x: EnvelopeTopicsData) -> Self {
+		EnvelopeTopic(x)
 	}
 }
 
-impl Topic {
-	/// set up to three bits in the 64-byte bloom passed.
+impl EnvelopeTopic {
+	/// Set up to three bits in the 64-byte bloom passed.
 	///
-	/// this takes 3 sets of 9 bits, treating each as an index in the range
+	/// This takes 3 sets of 9 bits, treating each as an index in the range
 	/// 0..512 into the bloom and setting the corresponding bit in the bloom to 1.
-	pub fn bloom_into(&self, bloom: &mut H512) {
+	pub fn bloom_into(&self, bloom: &mut Bloom) {
 
-		let data = &self.0;
+		let topics_data = &self.0;
 		for i in 0..3 {
-			let mut idx = data[i] as usize;
+			let mut topic_idx = topics_data[i] as BloomTopicIndex;
 
-			if data[3] & (1 << i) != 0 {
-				idx += 256;
+			// What does `1 << i` refer to
```

Comment:
1 << i:   is the same 2^i 

so 2^0, 2^1, 2^2.. 2^n .. -> 1, 2, 4, 8,,

Reply:
I for one decidedly prefer using constants or explicit numbers whenever possible.